downloader/artifact_downloader.go

```diff
@@ -145,22 +145,36 @@ func (ad *ConcurrentArtifactDownloader) downloadFile(targetDir, fileName, downlo
 
 		cancel()
 
+		start = time.Now()
+
 		ctx, cancel = context.WithTimeout(context.Background(), ad.Timeout)
 		downloader := got.NewWithContext(ctx)
 		downloader.Client = ad.createClient().StandardClient()
 		err = downloader.Download(downloadURL, fileFullPath)
 	}
 
 	if err != nil {
-		if err.Error() == "Response status code is not ok: 416" { // fallback to single threaded download - this error seems to happen for 0 size files with got
+		// fallback to single threaded download - the error with the 416 status code seems to happen for 0 size files with got
+		errorMessage := err.Error()
+		if errorMessage == "Response status code is not ok: 416" || strings.Contains(errorMessage, "unexpected EOF") {
```

Comment:
It does not do any harm if we try to execute a single threaded download in case of an `unexpected EOF` error. This download might be slower but at least the workflow will have the chance to succeed.